### ISSUE TITLE
无头浏览器Puppeteer

### ISSUE THREAD

**Issue by chenxiaochun** [chenxiaochun/blog#38]
https://github.com/GoogleChrome/puppeteer

**Comment by akeyboardlife** [comment]
每次滚动100有点慢，我直接让每次滚动document.body.scrollHeight，同时调高了时间间隔。

感谢你的代码。

**Comment by chenxiaochun** [comment]
@akeyboardlife ，有帮助就好，客气了！

**Comment by Asher-Tan** [comment]
赞！

**Comment by YiQieSuiYuan** [comment]
终端请求少了一行代码否则不生效
await page.setRequestInterception(true);
    page.on('request', interceptedRequest => {
        let url = interceptedRequest.url();
        if (url.indexOf('mmbiz.qpic.cn') > -1) {
            interceptedRequest.abort();
            // console.log(url);
        } else {
            interceptedRequest.continue();
        } 
        
    });

**Comment by ycjcl868** [comment]
window 对象怎么来的？node 里不存在这个

**Comment by leafney** [comment]
@YiQieSuiYuan Yes, it works well after adding `await page.setRequestInterception(true);` ,Thanks!

**Comment by yifeikong** [comment]
请问可以转载么，会注明出处和链接

**Comment by chenxiaochun** [comment]
@yifeikong ，可以的。

**Comment by yangweijie** [comment]
如何模拟设置下拉列表的值，甚至下拉列表是联动ajax关联请求的

**Comment by chenxiaochun** [comment]
@yangweijie ，这个得具体情况具体分析。如果是类似于地址关联的下拉框。可以等前面的下拉框变动之后，等待2秒钟，再去设置后面的下拉框值。这些在 puppeteer中都是有相关api的，可以查看一下。

**Comment by yangweijie** [comment]
> @yangweijie ，这个得具体情况具体分析。如果是类似于地址关联的下拉框。可以等前面的下拉框变动之后，等待2秒钟，再去设置后面的下拉框值。这些在 puppeteer中都是有相关api的，可以查看一下。

问题就不知道怎么让前面的下拉框变动，keyboard.type无效，要写额外js设置值吗？ 有的下拉可能 是其他js框架渲染的， 比如 这一个注册界面的 省市 下拉 http://interactive.cponline.cnipa.gov.cn/app/03_jh/login/register-qiye.jsp

**Comment by chenxiaochun** [comment]
@yangweijie ，我其实也好久不看它的文档了。刚才翻了一下文档，看到这个方法：https://pptr.dev/#?product=Puppeteer&version=v1.18.1&show=api-pageselectselector-values，估计可能对你有用。

**Comment by MBearo** [comment]
大佬有遇到截出来的图是空白么？用了你提供的方法，查了半天也不知道从何下手

**Comment by MBearo** [comment]
> 大佬有遇到截出来的图是空白么？用了你提供的方法，查了半天也不知道从何下手

把 headless 设置为 true 就好了，感谢感谢

**Comment by v2x2** [comment]
我尝试了设置了headless为true,还有滚动截图和fullPage: true，但是我截出来的图还是有白色的一段，我截的图片是偏大的有10000*20000，请问有什么方法可以解决吗？

**Comment by chenxiaochun** [comment]
> 我尝试了设置了headless为true,还有滚动截图和fullPage: true，但是我截出来的图还是有白色的一段，我截的图片是偏大的有10000*20000，请问有什么方法可以解决吗？

估计是图片太大，在截屏的时候还没有完全加载出来
